### PR TITLE
Using PAPI for e2e tests

### PR DESCRIPTION
**Pull request by mutantcornholio**
* Currently, e2e types are generated on postinstall and prebuild steps,
which requires metadata to be available in Docker image, even though
it's not technically used in the application code. I think separating
things would make things more complicated, but it's possible.
* Couldn't undestand the reason for having two different COPY instructions
in Dockerfile, but that doesn't live well with generating types in
postinstall, so merging those two
* .scale metadata files from Zombienet hosts saved in the codebase,
in order to be able to compile the code separately from running Zombienet

### CODE REVIEW

**Comment by rzadp** [comment]
> Couldn't undestand the reason for having two different COPY instructions
in Dockerfile

This is an optimization technique, to cache a docker layer with `node_modules` which doesn't get rebuilt when source files get changed.

**Comment by mutantcornholio** [comment]
> > Couldn't undestand the reason for having two different COPY instructions
> > in Dockerfile
> 
> This is an optimization technique, to cache a docker layer with `node_modules` which doesn't get rebuilt when source files get changed.

I see. After thinking more on this, decided to take this change back and go with another route.

File: e2e/README.md
Comment (rzadp) [inline e2e/README.md:1]: Can we add a note about how to obtain the `*.scale` files?

**Review by rzadp — APPROVED** [review]
🚀

**Review by Bullrich — APPROVED** [review]
LGTM

File: e2e/README.md
Comment (mutantcornholio) [reply e2e/README.md:1]: Good point, added

**Review by alvicsam — APPROVED** [review]

**Pull request merged**